### PR TITLE
Fix using tmp venv with NNCF installed in tests

### PR DESCRIPTION
**Pull request by vshampor**

### CODE REVIEW

**Review by aleksei-kashapov — COMMENTED** [review]
Why don't you like the name?

File: tests/conftest.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def ov_config_dir(request):
 
 
 @pytest.fixture(scope="function")
-def create_tmp_venv_with_nncf(install_type, tmp_path, package_type, venv_type):  # pylint:disable=redefined-outer-name
+def tmp_venv_with_nncf(install_type, tmp_path, package_type, venv_type):  # pylint:disable=redefined-outer-name
```
Comment (aleksei-kashapov) [inline tests/conftest.py:148]: Why don't you like the name?

File: tests/test_extensions_build.py [hunk 2]
```diff
@@ -10,13 +10,13 @@
...
                          [('venv', 'develop', 'GPU')])
-def test_force_cuda_build(create_tmp_venv_with_nncf, install_type, tmp_path, package_type):
+def test_force_cuda_build(tmp_venv_with_nncf, install_type, tmp_path, package_type):
     '''
     Check that CUDA Extensions weren't initially built and \
     then with TORCH_CUDA_ARCH_LIST were forced to be built
     '''
 
-    venv_path = tmp_path / 'venv'
+    venv_path = tmp_venv_with_nncf
```
Comment (aleksei-kashapov) [inline tests/test_extensions_build.py:19]: Anyway we have `tmp_path` as input. It is better to use only one way `tmp_path` or return value of `tmp_venv_with_nncf`

File: tests/conftest.py (hunk 1 above)
Comment (vshampor) [reply tests/conftest.py:148]: The built-in fixtures such as `tmp_path` and `tmp_dir` do not have `create` in them, so I decided to go for consistency here.

File: tests/test_extensions_build.py (hunk 2 above)
Comment (vshampor) [reply tests/test_extensions_build.py:19]: The tmp path with venv and the tmp path with the checks scripts should not necessarily be the same. In fact, it is even beneficial this way - less chance of common PYTHONPATH influence.

**Review by aleksei-kashapov — APPROVED** [review]

**Pull request merged**